server/app/controllers/files.js

```diff
@@ -0,0 +1,34 @@
+const uuid = require('uuid/v4');
+const AWS = require('aws-sdk');
+const asyncHandler = require('../middleware/async');
+
+const s3 = new AWS.S3({
+  accessKeyId: process.env.AWS_ACCESS_KEY,
+  secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
+});
+
+const { BUCKET_NAME, UPLOAD_DIRNAME } = process.env;
+
+const upload = asyncHandler(async (req, res, next) => {
+  if (!req.files || !req.files.file) throw Error('File is undefined');
```

Comment:
Instead of throwing an error, I created an ErrorResponse class that you can pass to the next() method. 
``
return next(new ErrorResponse('File is undefined', 400))
``
You can pass an error message and the status code.